pkg/inject/sidecar.go

```diff
@@ -177,3 +182,17 @@ func hasEnv(name string, vars []corev1.EnvVar) bool {
 	}
 	return false
 }
+
+// CleanSidecars of  deployments  associated with the jaeger instance.
+func CleanSidecars(deployments []appsv1.Deployment) {
+	for i := 0; i < len(deployments); i++ {
+		dep := &deployments[i]
+		for c := 0; c < len(dep.Spec.Template.Spec.Containers); c++ {
+			if dep.Spec.Template.Spec.Containers[c].Name == "jaeger-agent" {
+				// delete jaeger-agent container
+				dep.Spec.Template.Spec.Containers = append(dep.Spec.Template.Spec.Containers[:c], dep.Spec.Template.Spec.Containers[c+1:]...)
```

Comment:
Don't forget to remove the annotation as well.

Reply:
oh, you're right, PR updated with this! Thanks